plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -474,7 +479,14 @@ private fun SolverState.checkCallExpression(
                 solver.makeIntegerObjectVariable(associatedVarName)
             val arg1 = solver.makeIntegerObjectVariable(argVars[0].second)
             val arg2 = solver.makeIntegerObjectVariable(argVars[1].second)
-            specialCase(result, arg1, arg2)?.let { formula -> addConstraint(formula) }
+            specialCase(result, arg1, arg2)?.let { formula ->
+              addConstraint(
+                NamedConstraint(
+                  "checkCallArguments(${resolvedCall.resultingDescriptor.fqNameSafe}) [$result, $arg1, $arg2]",
```

Comment:
In theory we can already remove the `specialCase` function, maybe it's time to do so!

Reply:
To do this we need to figure out a way to depend on the laws module and add a small prelude for the ones there. Let's do that in a different PR